lms/validation/_base.py

```diff
@@ -69,6 +69,9 @@ def __init__(self, request):
         super().__init__()
         self.context = {"request": request}
 
+        # Custom "locations" for subclasses of PyramidRequestSchema
+        pyramidparser.PyramidParser.__location_map__["query_and_form"] = _query_and_form
```

Comment:
We are building a new `pyramidparser.PyramidParser` that takes a location so will modify the class directly here.

Reply:
I don't think we have to reach inside the class to get this done. Looking at the docs they seem to provide a way to get it done with a decorator:

 * https://webargs.readthedocs.io/en/latest/advanced.html#meta-locations

The example they give is actually `query_and_form`!